cider/exec/module/CiderRuntimeModule.cpp

```diff
@@ -734,24 +741,29 @@ CiderBatch CiderRuntimeModule::setSchemaAndUpdateCountDistinctResIfNeed(
 
   if (ciderCompilationResult_->impl_->query_mem_desc_->getQueryDescriptionType() ==
       QueryDescriptionType::NonGroupedAggregate) {
-    // for non-groupby partial avg, sum(int) may has double output Type thus
-    // need cast original int value(based on cider rule) to double
+    // For non-groupby partial avg, the output type of SUM(integer-type col) must be
```

Comment:
my understanding here is the output type is decided by caller side generated plan. So Cider will not have a strong behavior here. in other words, only specific output types are supported.

Reply:
yep, we change actual output data here to meet the requirement of Velox and there is only int64 which is needed to do this cast.